tensorflow_datasets/image_classification/pklot/pklot.py

```diff
@@ -0,0 +1,100 @@
+"""pklot dataset."""
+
+import os
+from itertools import chain
+from pathlib import Path
+
+import tensorflow_datasets.public_api as tfds
+
+_DESCRIPTION = """
+This database contains 12,417 images (1280X720) captured from two different parking lots (parking1 and parking2) in sunny, cloudy and rainy days. The first parking lot has two different capture angles (parking1a and parking 1b).
+
+The images are organised into three directories (parking1a, parking1b and parking2). Each directory contains three subdirectories for different weather conditions (cloudy, rainy and sunny). Inside of each subdirectory the images are organised by acquisition date.
+
+Each image of the database has a XML file associated including the coordinates of all the parking spaces and its label (occupied/vacant). By using the XML files to segment the parking space, you will be able to get around 695,900 images of parking spaces.
+
+More info about the database can be found in this readme file.
+"""
+
+_CITATION = """
+Almeida, P., Oliveira, L. S., Silva Jr, E., Britto Jr, A., Koerich, A., PKLot - A robust dataset for parking lot classification, Expert Systems with Applications, 42(11):4937-4949, 2015.
+"""
+
+_EMPTY_LABEL = "Empty"
+_OCCUPIED_LABEL = "Occupied"
+_EMPTY_LABEL_DIR_NAME = _EMPTY_LABEL
+_OCCUPIED_LABEL_DIR_NAME = _OCCUPIED_LABEL
+_LABEL_DIR_NAME_TO_LABEL_MAP = {
+    _EMPTY_LABEL_DIR_NAME: _EMPTY_LABEL,
+    _OCCUPIED_LABEL_DIR_NAME: _OCCUPIED_LABEL
+}
+
+
+class Pklot(tfds.core.GeneratorBasedBuilder):
+  """DatasetBuilder for pklot dataset."""

```

Comment:
The dataset provides original and segmented images that have a slightly different directory structure:
https://www.tensorflow.org/datasets/add_dataset?hl=en#dataset_configurationvariants_tfdscorebuilderconfig

Reply:
I just saw GitHub thanks a ton for reviewing this!, Will get to it asap 🙏